022-quarkus-properties-config-all/src/main/resources/application.properties

```diff
@@ -3,4 +3,20 @@ antagonist.message=Hi, I am Sheldon Plankton
 antagonist.wife.name=Karen
 antagonist.wife.message=Hi, I am Karen
 
-restaurant.employees=WillBeOverridden,WillBeOverridden,WillBeOverridden
\ No newline at end of file
+restaurant.employees=WillBeOverridden,WillBeOverridden,WillBeOverridden
+
+server.url=http://${server.host}/endpoint
+server.host=example.org
+server.path=mypath
+server.port=8080
+
+# default properties
+server.url.with.default.found=http://${server.host:default.org}/endpoint
+server.url.with.default.not.found=http://${server.host.not.found:default.org}/endpoint
+server.url.with.default.nested=http://${server.host.not.found:${server.host}}/endpoint
+
+# composed properties
+server.url.composed=http://${server.host}:${server.port}/${server.path}
+
+# no substitution
+server.url.raw=\\${server.host}
```

Comment:
Please add a comment that these properties target .bulk package. So separation of properties is clear

Reply:
As discussed, I prefer keeping the properties agnostic from where will be used now and/or in the future.